File: 12-SpotifyAPI/SpotifyTest/SpotifyTest/Artist.m
//
//  Artist.m
//  SpotifyTest
//
//  Created by Phil Wright on 6/7/16.
//  Copyright © 2016 Touchopia, LLC. All rights reserved.
//

#import "Artist.h"

@implementation Artist

-(instancetype) initWithDictionary:(NSDictionary *)dict {
    
    if ( (self = [super init])) {
        //NSLog(@"dict == %@",dict);
        
        //NSLog(@"name == %@",[itemDict valueForKey:@"name"]);
        
        self.name = [dict valueForKey:@"name"];
        
        //NSLog(@"id == %@",[itemDict valueForKey:@"id"]);
        
        self.artistId = [dict valueForKey:@"id"];
        
        
        // get the images array
        
        NSArray *imagesArray = [dict valueForKey:@"images"];
        
    }
    return self;
}


@end

File: 20-Spotify-Top-Tracks/TopTracksApp/TopTracksApp/Artist.swift
//
//  Artist.swift
//  TopTracksApp
//
//  Created by Phil Wright on 6/20/16.
//  Copyright © 2016 Touchopia, LLC. All rights reserved.
//

import UIKit

class Artist: NSObject {

    var name: String = ""
    var artistID: String = ""
    
    override init() {
        super.init()
        self.name = ""
        self.artistID = ""
    }
    
    init(dict: JSONDictionary) {
        
        super.init()
        
        if let name = dict["name"] as? String {
            self.name = name
        } else {
            print("I could not parse the name")
        }
        
        if let artistID = dict["id"] as? String {
            self.artistID = artistID
            
        } else {
            print("I could not parse the id")
        }
        
    }
    
}
